process/block/baseProcess.go

```diff
@@ -1011,10 +1015,14 @@ func (bp *baseProcessor) cleanupPoolsForCrossShard(
 		crossNotarizedHeader.GetNonce(),
 	)
 
-	err = bp.dataPool.Proofs().CleanupProofsBehindNonce(shardID, noncesToPrevFinal)
-	if err != nil {
-		log.Warn("%w: failed to cleanup notarized proofs behind nonce %d on shardID %d",
-			err, noncesToPrevFinal, shardID)
+	if bp.enableEpochsHandler.IsFlagEnabled(common.EquivalentMessagesFlag) {
```

Comment:
```suggestion
	if bp.enableEpochsHandler.IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, crossNotarizedHeader.GetEpoch()) {
```

not sure if this would be the proper epoch to check against